pydantic/class_validators.py

```diff
@@ -78,8 +80,9 @@ def validator(
         each_item = not whole
 
     def dec(f: AnyCallable) -> classmethod:
-        _check_validator_name(f)
-        f_cls = classmethod(f)
+        f_cls = classmethod(f) if not isinstance(f, classmethod) else f
```

Comment:
```suggestion
        f_cls = f if isinstance(f, classmethod) else classmethod(f)
```
No need to use a `not`.

Reply:
Also below. 

But we do the same thing 3 times here, could we move this logic into `_check_validator_name`? Perhaps renamed to 

```py
def _prepare_validator(f: Any, allow_reuse: bool = False) -> classmethod:
    ...
```